### ISSUE TITLE
PowerShell - Copy-Item: Cannot overwrite the item K:\file.jpg with itself

### ISSUE THREAD

**Issue by KyleKolander** [winfsp/winfsp#564]
## Bug Report

When running the passthrough-dotnet sample app as follows, PowerShell 7 Copy-Item commands don't work.

`.\passthrough-dotnet.exe -p "P:\Web2" -m "K:" -d -1 -D -`

### How to Reproduce

Here's a tree output of the mounted K: drive (it's the same for the source P:\Web2):

```
K:\
|-- test
`-- file.jpg

1 directory, 1 file
```

Open a PowerShell 7 prompt and run this command: `Copy-Item -Path K:\file.jpg -Destination K:\test\`

### Behaviors

Expected behavior is that the copy operation will work.  😏 

Running this from a Cmd prompt works as expected: `cp K:\file.jpg K:\test\`

Copying the file with File Explorer works as expected, too.

Move-Item works in PowerShell.  It just seems to be the Copy-Item that won't work.

### Environment

Windows 11 Pro Version 23H2 (OS Build 22631.3810)
Windows Feature Experience Pack 1000.22700.1020.0
Windows Terminal Preview Version: 1.21.1772.0
PowerShell 7.4.3

**Comment by billziss-gh** [comment]
Does this problem happen with MEMFS or NTPTFS?

The passthrough file systems were intended as tutorial file systems and as such they may not support all file system scenarios.

**Comment by KyleKolander** [comment]
Yes, it works on memfs-dotnet.  But I'm needing to use an NTFS source, not in-memory.  Are you able to replicate the bug?

**Comment by billziss-gh** [comment]
I am currently on vacation and without access to a Windows computer.

Please keep in mind that MEMFS and NTPTFS are the WinFsp reference file systems. Other file systems are educational samples and any problems may not get fixed.

**Comment by ubraig** [comment]
Same issue here:
- PowerShell 7.4.6 on Windows 10
- Cryptomator 1.14.2
- Mounted a vault to T:\ with "WinFSP (LocalDrive)" or "WinFSP"

Steps to reproduce on a powershell command line:
```
cd T:\
md A
md B
set-content a\test.txt "Hello World"
```

`Copy-Item .\a\test.txt .\b\ `
--> **Copy-Item: Cannot overwrite the item T:\a\test.txt with itself.**

`Copy-Item .\a\test.txt .\b\test.txt`
--> works fine

However, on any plain NTFS drive (without cryptomator), both commands work fine which is the expected behaviour of the Copy-Item command

**Comment by billziss-gh** [comment]
@ubraig

In order for me to consider something as a potential problem with WinFsp, I need to be able to replicate it on MEMFS. The reason is that MEMFS is the reference file system for WinFsp, which I strive to keep compatible with NTFS. If something does not work on MEMFS, then there is a good chance that it is a problem with WinFsp in general.

I will also accept bug reports against NTPTFS.

I cannot accept reports against other file systems, because I have found that too many times the problems were with the file systems rather than WinFsp.

**Comment by ubraig** [comment]
@billziss-gh  Ok, then this actually means, that Cryptomator is incomatible with and not supported on Windows...
